src/IO/H5/ContiguousVolumeData.cpp

```diff
@@ -0,0 +1,250 @@
+              local_connectivity.emplace_back(
+                  point_counter + static_cast<int>(bounding_indices));
+            }
+          }
+          return local_connectivity;
+        }
+        ();
+        point_counter += local_numpoints;
+        total_connectivity.insert(total_connectivity.end(),
+                                  connectivity.begin(), connectivity.end());
+
+      }  // if (First)
+      auto tensor_component = element.tensor_components[i];
+      DataVector local_tensor_data = tensor_component.data;
+      tensor_data.insert(tensor_data.end(), local_tensor_data.begin(),
+                         local_tensor_data.end());
+
+    }  // for each element
+
+    // Write the data for the tensor component
+    std::vector<size_t> Size{tensor_data.size()};
+    h5::write_data(observation_group.id(), tensor_data, Size, component_name);
+    First = false;
+  }  // for each component
+
+  // Write the grid extents contiguously, the first `dim` belong to the
+  // First grid, the second `dim` belong to the second grid, and so on,
+  // Ordering is `x, y, z, ... `
+  h5::write_data(observation_group.id(), total_extents,
+                 std::vector<size_t>{total_extents.size()}, "total_extents");
+  // Write the names of the grids as vector of chars with individual names
+  // separated by colons
+  std::vector<size_t> names_length{grid_names.length()};
+  std::vector<char> charvec(grid_names.begin(), grid_names.end());
+
+  h5::write_data(observation_group.id(), charvec, names_length, "grid_names");
+  // Write the Connectivity
+  std::vector<size_t> connectivity_length{total_connectivity.size()};
+  h5::write_data(observation_group.id(), total_connectivity,
+                 connectivity_length, "connectivity");
+}
+
+std::vector<size_t> ContiguousVolumeData::list_observation_ids() const
+    noexcept {
+  const auto names = get_group_names(volume_data_group_.id(), "");
+  const auto helper = [](const std::string& s) noexcept {
+    return std::stoul(s.substr(std::string("ObservationId").size()));
+  };
+  return {boost::make_transform_iterator(names.begin(), helper),
+          boost::make_transform_iterator(names.end(), helper)};
+}
+
+double ContiguousVolumeData::get_observation_value(
+    const size_t observation_id) const noexcept {
+  const std::string path = "ObservationId" + std::to_string(observation_id);
+  detail::OpenGroup observation_group(volume_data_group_.id(), path,
+                                      AccessType::ReadOnly);
+  return h5::read_value_attribute<double>(observation_group.id(),
+                                          "observation_value");
+}
+
+std::vector<std::string> ContiguousVolumeData::list_tensor_components(
+    size_t observation_id) const noexcept {
+  auto tensor_components =
+      get_group_names(volume_data_group_.id(),
+                      "ObservationId" + std::to_string(observation_id));
+  std::remove(tensor_components.begin(), tensor_components.end(),
```

Comment:
`// std::remove moves the element to the end of the vector, so we still need to actually erase it from the vector`